<!DOCTYPE html>
<html>

<head>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>grm_chatgpt</title>


<style type="text/css">
body {
  font-family: Helvetica, arial, sans-serif;
  font-size: 14px;
  line-height: 1.6;
  padding-top: 10px;
  padding-bottom: 10px;
  background-color: white;
  padding: 30px; }

body > *:first-child {
  margin-top: 0 !important; }
body > *:last-child {
  margin-bottom: 0 !important; }

a {
  color: #4183C4; }
a.absent {
  color: #cc0000; }
a.anchor {
  display: block;
  padding-left: 30px;
  margin-left: -30px;
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0; }

h1, h2, h3, h4, h5, h6 {
  margin: 20px 0 10px;
  padding: 0;
  font-weight: bold;
  -webkit-font-smoothing: antialiased;
  cursor: text;
  position: relative; }

h1:hover a.anchor, h2:hover a.anchor, h3:hover a.anchor, h4:hover a.anchor, h5:hover a.anchor, h6:hover a.anchor {
  background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA09pVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMy1jMDExIDY2LjE0NTY2MSwgMjAxMi8wMi8wNi0xNDo1NjoyNyAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNiAoMTMuMCAyMDEyMDMwNS5tLjQxNSAyMDEyLzAzLzA1OjIxOjAwOjAwKSAgKE1hY2ludG9zaCkiIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OUM2NjlDQjI4ODBGMTFFMTg1ODlEODNERDJBRjUwQTQiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OUM2NjlDQjM4ODBGMTFFMTg1ODlEODNERDJBRjUwQTQiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo5QzY2OUNCMDg4MEYxMUUxODU4OUQ4M0REMkFGNTBBNCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo5QzY2OUNCMTg4MEYxMUUxODU4OUQ4M0REMkFGNTBBNCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PsQhXeAAAABfSURBVHjaYvz//z8DJYCRUgMYQAbAMBQIAvEqkBQWXI6sHqwHiwG70TTBxGaiWwjCTGgOUgJiF1J8wMRAIUA34B4Q76HUBelAfJYSA0CuMIEaRP8wGIkGMA54bgQIMACAmkXJi0hKJQAAAABJRU5ErkJggg==) no-repeat 10px center;
  text-decoration: none; }

h1 tt, h1 code {
  font-size: inherit; }

h2 tt, h2 code {
  font-size: inherit; }

h3 tt, h3 code {
  font-size: inherit; }

h4 tt, h4 code {
  font-size: inherit; }

h5 tt, h5 code {
  font-size: inherit; }

h6 tt, h6 code {
  font-size: inherit; }

h1 {
  font-size: 28px;
  color: black; }

h2 {
  font-size: 24px;
  border-bottom: 1px solid #cccccc;
  color: black; }

h3 {
  font-size: 18px; }

h4 {
  font-size: 16px; }

h5 {
  font-size: 14px; }

h6 {
  color: #777777;
  font-size: 14px; }

p, blockquote, ul, ol, dl, li, table, pre {
  margin: 15px 0; }

hr {
  background: transparent url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAYAAAAECAYAAACtBE5DAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAyJpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNSBNYWNpbnRvc2giIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OENDRjNBN0E2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OENDRjNBN0I2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo4Q0NGM0E3ODY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo4Q0NGM0E3OTY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PqqezsUAAAAfSURBVHjaYmRABcYwBiM2QSA4y4hNEKYDQxAEAAIMAHNGAzhkPOlYAAAAAElFTkSuQmCC) repeat-x 0 0;
  border: 0 none;
  color: #cccccc;
  height: 4px;
  padding: 0;
}

body > h2:first-child {
  margin-top: 0;
  padding-top: 0; }
body > h1:first-child {
  margin-top: 0;
  padding-top: 0; }
  body > h1:first-child + h2 {
    margin-top: 0;
    padding-top: 0; }
body > h3:first-child, body > h4:first-child, body > h5:first-child, body > h6:first-child {
  margin-top: 0;
  padding-top: 0; }

a:first-child h1, a:first-child h2, a:first-child h3, a:first-child h4, a:first-child h5, a:first-child h6 {
  margin-top: 0;
  padding-top: 0; }

h1 p, h2 p, h3 p, h4 p, h5 p, h6 p {
  margin-top: 0; }

li p.first {
  display: inline-block; }
li {
  margin: 0; }
ul, ol {
  padding-left: 30px; }

ul :first-child, ol :first-child {
  margin-top: 0; }

dl {
  padding: 0; }
  dl dt {
    font-size: 14px;
    font-weight: bold;
    font-style: italic;
    padding: 0;
    margin: 15px 0 5px; }
    dl dt:first-child {
      padding: 0; }
    dl dt > :first-child {
      margin-top: 0; }
    dl dt > :last-child {
      margin-bottom: 0; }
  dl dd {
    margin: 0 0 15px;
    padding: 0 15px; }
    dl dd > :first-child {
      margin-top: 0; }
    dl dd > :last-child {
      margin-bottom: 0; }

blockquote {
  border-left: 4px solid #dddddd;
  padding: 0 15px;
  color: #777777; }
  blockquote > :first-child {
    margin-top: 0; }
  blockquote > :last-child {
    margin-bottom: 0; }

table {
  padding: 0;border-collapse: collapse; }
  table tr {
    border-top: 1px solid #cccccc;
    background-color: white;
    margin: 0;
    padding: 0; }
    table tr:nth-child(2n) {
      background-color: #f8f8f8; }
    table tr th {
      font-weight: bold;
      border: 1px solid #cccccc;
      margin: 0;
      padding: 6px 13px; }
    table tr td {
      border: 1px solid #cccccc;
      margin: 0;
      padding: 6px 13px; }
    table tr th :first-child, table tr td :first-child {
      margin-top: 0; }
    table tr th :last-child, table tr td :last-child {
      margin-bottom: 0; }

img {
  max-width: 100%; }

span.frame {
  display: block;
  overflow: hidden; }
  span.frame > span {
    border: 1px solid #dddddd;
    display: block;
    float: left;
    overflow: hidden;
    margin: 13px 0 0;
    padding: 7px;
    width: auto; }
  span.frame span img {
    display: block;
    float: left; }
  span.frame span span {
    clear: both;
    color: #333333;
    display: block;
    padding: 5px 0 0; }
span.align-center {
  display: block;
  overflow: hidden;
  clear: both; }
  span.align-center > span {
    display: block;
    overflow: hidden;
    margin: 13px auto 0;
    text-align: center; }
  span.align-center span img {
    margin: 0 auto;
    text-align: center; }
span.align-right {
  display: block;
  overflow: hidden;
  clear: both; }
  span.align-right > span {
    display: block;
    overflow: hidden;
    margin: 13px 0 0;
    text-align: right; }
  span.align-right span img {
    margin: 0;
    text-align: right; }
span.float-left {
  display: block;
  margin-right: 13px;
  overflow: hidden;
  float: left; }
  span.float-left span {
    margin: 13px 0 0; }
span.float-right {
  display: block;
  margin-left: 13px;
  overflow: hidden;
  float: right; }
  span.float-right > span {
    display: block;
    overflow: hidden;
    margin: 13px auto 0;
    text-align: right; }

code, tt {
  margin: 0 2px;
  padding: 0 5px;
  white-space: nowrap;
  border: 1px solid #eaeaea;
  background-color: #f8f8f8;
  border-radius: 3px; }

pre code {
  margin: 0;
  padding: 0;
  white-space: pre;
  border: none;
  background: transparent; }

.highlight pre {
  background-color: #f8f8f8;
  border: 1px solid #cccccc;
  font-size: 13px;
  line-height: 19px;
  overflow: auto;
  padding: 6px 10px;
  border-radius: 3px; }

pre {
  background-color: #f8f8f8;
  border: 1px solid #cccccc;
  font-size: 13px;
  line-height: 19px;
  overflow: auto;
  padding: 6px 10px;
  border-radius: 3px; }
  pre code, pre tt {
    background-color: transparent;
    border: none; }

sup {
    font-size: 0.83em;
    vertical-align: super;
    line-height: 0;
}

kbd {
  display: inline-block;
  padding: 3px 5px;
  font-size: 11px;
  line-height: 10px;
  color: #555;
  vertical-align: middle;
  background-color: #fcfcfc;
  border: solid 1px #ccc;
  border-bottom-color: #bbb;
  border-radius: 3px;
  box-shadow: inset 0 -1px 0 #bbb
}

* {
	-webkit-print-color-adjust: exact;
}
@media screen and (min-width: 914px) {
    body {
        width: 854px;
        margin:0 auto;
    }
}
@media print {
	table, pre {
		page-break-inside: avoid;
	}
	pre {
		word-wrap: break-word;
	}
  body {
    padding: 2cm; 
  }
}
</style>

<style type="text/css">
/**
 * prism.js default theme for JavaScript, CSS and HTML
 * Based on dabblet (http://dabblet.com)
 * @author Lea Verou
 */

code[class*="language-"],
pre[class*="language-"] {
	color: black;
	background: none;
	text-shadow: 0 1px white;
	font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
	text-align: left;
	white-space: pre;
	word-spacing: normal;
	word-break: normal;
	word-wrap: normal;
	line-height: 1.5;

	-moz-tab-size: 4;
	-o-tab-size: 4;
	tab-size: 4;

	-webkit-hyphens: none;
	-moz-hyphens: none;
	-ms-hyphens: none;
	hyphens: none;
}

pre[class*="language-"]::-moz-selection, pre[class*="language-"] ::-moz-selection,
code[class*="language-"]::-moz-selection, code[class*="language-"] ::-moz-selection {
	text-shadow: none;
	background: #b3d4fc;
}

pre[class*="language-"]::selection, pre[class*="language-"] ::selection,
code[class*="language-"]::selection, code[class*="language-"] ::selection {
	text-shadow: none;
	background: #b3d4fc;
}

@media print {
	code[class*="language-"],
	pre[class*="language-"] {
		text-shadow: none;
	}
}

/* Code blocks */
pre[class*="language-"] {
	padding: 1em;
	margin: .5em 0;
	overflow: auto;
}

:not(pre) > code[class*="language-"],
pre[class*="language-"] {
	background: #f5f2f0;
}

/* Inline code */
:not(pre) > code[class*="language-"] {
	padding: .1em;
	border-radius: .3em;
	white-space: normal;
}

.token.comment,
.token.prolog,
.token.doctype,
.token.cdata {
	color: slategray;
}

.token.punctuation {
	color: #999;
}

.namespace {
	opacity: .7;
}

.token.property,
.token.tag,
.token.boolean,
.token.number,
.token.constant,
.token.symbol,
.token.deleted {
	color: #905;
}

.token.selector,
.token.attr-name,
.token.string,
.token.char,
.token.builtin,
.token.inserted {
	color: #690;
}

.token.operator,
.token.entity,
.token.url,
.language-css .token.string,
.style .token.string {
	color: #a67f59;
	background: hsla(0, 0%, 100%, .5);
}

.token.atrule,
.token.attr-value,
.token.keyword {
	color: #07a;
}

.token.function {
	color: #DD4A68;
}

.token.regex,
.token.important,
.token.variable {
	color: #e90;
}

.token.important,
.token.bold {
	font-weight: bold;
}
.token.italic {
	font-style: italic;
}

.token.entity {
	cursor: help;
}
</style>


</head>

<body>

<h1 id="toc_0">ChatGPT Q&amp;A - Statistical Analysis for GWAS and Population Genetics</h1>

<h2 id="toc_1">Shweta Pipaliya - February 22, 2023</h2>

<h3 id="toc_2">Q1. I want to perform GRM analysis using PLINK input format files. How should I run this using GCTA</h3>

<blockquote>
<ul>
<li><p>Performing a genome-wide relatedness matrix (GRM) analysis using GCTA requires the following steps:</p></li>
<li><p>Convert your PLINK input files into GCTA format using the plink2gcta command in GCTA. For example:</p></li>
</ul>

<div><pre><code class="language-none">gcta64 --plink2gcta --bfile mydata --make-grm --out mydata</code></pre></div>

<ul>
<li><p>This command will create a GCTA-format file mydata.grm.bin containing the GRM.</p></li>
<li><p>Use the grm command in GCTA to perform the GRM analysis. For example:</p></li>
</ul>

<div><pre><code class="language-none">gcta64 --grm mydata --grm-cutoff 0.05 --out mydata_grm</code></pre></div>

<ul>
<li><p>This command will calculate the genetic relationship matrix for your data and output the results to the file mydata_grm.</p></li>
<li><p>In the above example, --grm-cutoff 0.05 sets a cutoff for the minor allele frequency (MAF) at 5% for SNPs included in the GRM calculation. You can adjust this value as needed based on the characteristics of your dataset.</p></li>
<li><p>Additionally, there are other parameters you may want to use in the grm command, such as --thread-num to specify the number of threads to use for the calculation.</p></li>
</ul>
</blockquote>

<h3 id="toc_3">Q2. How can I use the genetic relationship matrix (grm) to perform a GWAS using mixed linear model in GCTA?</h3>

<blockquote>
<ul>
<li><p>You can use the genetic relationship matrix (GRM) obtained from GCTA to perform genome-wide association studies (GWAS) using a mixed linear model (MLM) approach in GCTA. Here are the steps to follow:</p></li>
<li><p>Prepare your phenotype and genotype data in GCTA format using the --pheno and --bfile options, respectively. For example:</p></li>
</ul>

<div><pre><code class="language-none">gcta64 --bfile mydata --pheno mypheno.txt --make-grm --out mydata</code></pre></div>

<ul>
<li><p>This command will create a GCTA-format file mydata.grm.bin containing the GRM and use the phenotype data in mypheno.txt for analysis.</p></li>
<li><p>Use the mlma command in GCTA to perform the MLM analysis. For example:</p></li>
</ul>

<div><pre><code class="language-none">gcta64 --mlma --grm mydata --pheno mypheno.txt --out myresult</code></pre></div>

<ul>
<li><p>This command will perform the MLM analysis using the GRM obtained from step 1 and output the results to the file myresult.</p></li>
<li><p>In the above example, you can also add the --thread-num option to specify the number of threads to use for the calculation.</p></li>
<li><p>The output file myresult will contain the following columns: SNP name, chromosome, position, effect allele frequency, beta coefficient, standard error, p-value, and significance level after Bonferroni correction.</p></li>
</ul>
</blockquote>

<h3 id="toc_4">Q3. Do I need to perform a principal component analysis and include the resulting PCs as covariates for an mlm GWAS?</h3>

<blockquote>
<ul>
<li><p>Performing a principal component analysis (PCA) and including the resulting principal components (PCs) as covariates in a mixed linear model (MLM) GWAS can help to control for population stratification and reduce the risk of false positive associations. However, the need for PCA depends on the characteristics of your dataset.</p></li>
<li><p>If your dataset is well-matched for population structure, and there is no significant population stratification, you may not need to perform PCA and include PCs as covariates. In such cases, using the GRM obtained from GCTA as a covariate in the MLM analysis may be sufficient to control for population structure.</p></li>
<li><p>On the other hand, if your dataset has significant population stratification, including the top PCs as covariates in the MLM analysis can help to adjust for this effect and improve the accuracy of the results.</p></li>
<li><p>In summary, whether or not to perform PCA and include PCs as covariates depends on the characteristics of your dataset, and it is a good practice to assess the population structure and perform appropriate adjustments to control for it.</p></li>
</ul>
</blockquote>

<h3 id="toc_5">Q4. Should I provide a covariate file to run an mlm GWAS using GCTA?</h3>

<blockquote>
<ul>
<li><p>Providing a covariate file is optional when running a mixed linear model (MLM) genome-wide association study (GWAS) using GCTA.</p></li>
<li><p>Including covariates in the MLM model can improve the accuracy of the results by adjusting for any confounding effects in the data. Covariates can include demographic variables, such as age and sex, or other relevant factors that may influence the phenotype of interest.</p></li>
<li><p>If you have covariate data that you believe may impact your GWAS results, you can provide a covariate file to GCTA using the --covar option, followed by the filename. For example:</p></li>
</ul>

<div><pre><code class="language-none">gcta64 --mlma --grm mydata --pheno mypheno.txt --covar mycovar.txt --out myresult</code></pre></div>

<ul>
<li><p>In this example, mycovar.txt contains the covariate data that will be included in the MLM model alongside the genetic relationship matrix (GRM) and phenotype data.</p></li>
<li><p>If you do not have any covariate data or do not wish to include covariates in your MLM analysis, you can simply omit the --covar option from your command line.</p></li>
<li><p>It is worth noting that the decision to include or exclude covariates should be based on a careful consideration of the characteristics of your dataset and the nature of the phenotype you are studying.</p></li>
</ul>
</blockquote>

<h3 id="toc_6">Q5. Can you perform a power calculation for a GWAS with sample size of 2200 and MAF threshold of less than 1% with a p-value significance threshold of 5x10^-8</h3>

<blockquote>
<ul>
<li><p>Yes, you can perform a power calculation for a GWAS with a sample size of 2200, MAF threshold of less than 1%, and a p-value significance threshold of 5x10^-8.</p></li>
<li><p>Here&#39;s an example of how you can perform the power calculation using G*Power software:</p>

<ul>
<li><p>Open G*Power software and select &quot;t tests&quot; from the test family menu.</p></li>
<li><p>Choose &quot;Logistic regression: Wald test&quot; as the test type.</p></li>
<li><p>Enter the following parameters:</p>

<ul>
<li>Effect size (odds ratio): assume a hypothetical odds ratio of 1.2</li>
<li>Significance level: 5x10^-8</li>
<li>Power: 80%</li>
<li>Sample size: 2200</li>
<li>Number of predictors (excluding the constant): assume 1 predictor (the SNP)</li>
</ul></li>
<li><p>Click the &quot;Calculate&quot; button to obtain the results.</p></li>
</ul></li>
<li><p>Based on the above parameters, G*Power calculates that a sample size of 2200 would have 80% power to detect an odds ratio of 1.2 at a significance level of 5x10^-8, assuming a SNP with MAF of less than 1%. However, please note that this calculation is based on several assumptions and may not be an accurate reflection of the actual power of your study.</p></li>
</ul>
</blockquote>

<h3 id="toc_7">Q6. Calculate the odds ratio for a GWAS with sample size of 1942, minor allele frequency threshold of &gt; 1%, and a genome wide significance threshold of p &gt; 5x10^-8. The number of controls are 1300 while cases are 642.</h3>

<blockquote>
<ul>
<li><p>To calculate the odds ratio for a GWAS, we first need to obtain the number of cases and controls who carry the minor allele and who do not. We can then use these counts to calculate the odds ratio.</p></li>
<li><p>Let&#39;s start by calculating the number of cases and controls who carry the minor allele. We know that the minor allele frequency threshold is &gt; 1%, so any variant with a frequency less than this threshold will be excluded from the analysis.</p></li>
<li><p>Assuming that there are no missing genotypes, we can use the following formula to calculate the number of cases and controls who carry the minor allele:</p></li>
<li><p>Number of cases with minor allele = number of cases * minor allele frequency
Number of controls with minor allele = number of controls * minor allele frequency</p></li>
<li><p>Plugging in the numbers given in the question, we get:</p></li>
<li><p>Number of cases with minor allele = 642 * 0.01 = 6.42</p></li>
<li><p>Number of controls with minor allele = 1300 * 0.01 = 13</p></li>
<li><p>Next, we can calculate the number of cases and controls who do not carry the minor allele:</p></li>
<li><p>Number of cases without minor allele = number of cases - number of cases with minor allele</p></li>
<li><p>Number of controls without minor allele = number of controls - number of controls with minor allele</p></li>
<li><p>Plugging in the numbers given in the question, we get:</p></li>
<li><p>Number of cases without minor allele = 642 - 6.42 = 635.58</p></li>
<li><p>Number of controls without minor allele = 1300 - 13 = 1287</p></li>
<li><p>Now we can use these counts to calculate the odds ratio. The odds ratio (OR) is the ratio of the odds of the outcome (i.e., disease) in the group exposed to the risk factor (i.e., minor allele) to the odds of the outcome in the group not exposed to the risk factor. The formula for calculating OR is:</p>

<ul>
<li><p>OR = (a/b) / (c/d)</p>

<ul>
<li>where a is the number of cases with the minor allele, b is the number of cases without the minor allele, c is the number of controls with the minor allele, and d is the number of controls without the minor allele.</li>
</ul></li>
</ul></li>
<li><p>Plugging in the numbers from above, we get:</p>

<ul>
<li>OR = (6.42/635.58) / (13/1287) = 1.218</li>
</ul></li>
<li><p>Therefore, the odds ratio for this GWAS is 1.218.</p></li>
</ul>
</blockquote>

<h3 id="toc_8">Q7. What is the difference between mlm-LOCO and mlm in GCTA?</h3>

<blockquote>
<ul>
<li><p>MLM-LOCO and MLM are two methods used in GCTA (Genome-wide Complex Trait Analysis) to perform genome-wide association studies (GWAS) and estimate the genetic effects of variants on complex traits. The main difference between the two methods lies in the way they control for population structure and relatedness between individuals.</p></li>
<li><p>MLM (Mixed Linear Model) is a method that uses a linear mixed model to account for population structure and relatedness in GWAS. It estimates the genetic effects of variants on a trait while accounting for the correlation structure of individuals in the study due to their relatedness and shared ancestry. The MLM method uses a genetic relationship matrix (GRM) to model the covariance structure of the genetic effects and the residuals, and estimates the variance components of the model using restricted maximum likelihood (REML).</p></li>
<li><p>MLM-LOCO (MLM with Leave-One-Chromosome-Out) is a variant of the MLM method that controls for population structure and relatedness by fitting a linear mixed model for each chromosome separately while leaving out the chromosome of interest. This method is particularly useful when the genetic effects of a variant on a trait are confounded by population structure or relatedness on a specific chromosome. By fitting the model on the remaining chromosomes, MLM-LOCO can remove the confounding effects of population structure and relatedness from the chromosome of interest, allowing for more accurate estimation of the genetic effects of variants.</p></li>
<li><p>In summary, while both MLM and MLM-LOCO use linear mixed models to account for population structure and relatedness, MLM-LOCO fits the model on each chromosome separately while leaving out the chromosome of interest. This can lead to more a
ccurate estimation of the genetic effects of variants in GWAS, particularly when population structure or relatedness is confounding the genetic effects on a specific chromosome.</p></li>
</ul>
</blockquote>

<h3 id="toc_9">Q8. What are the parameters to run a GWAS using MLM-LOCO?</h3>

<blockquote>
<ul>
<li><p>`To run a genome-wide association study (GWAS) using MLM-LOCO in GCTA (Genome-wide Complex Trait Analysis), you need to specify several parameters in the GCTA command line. Here are the main parameters required to run MLM-LOCO:</p></li>
<li><p>The input data file: This is a file containing the genotypes and phenotypes of the individuals in the study. The genotypes are typically represented as a binary PLINK file (.bed, .bim, .fam), while the phenotypes are represented as a text file (.phe).</p></li>
<li><p>The GRM file: This is a file containing the genetic relationship matrix (GRM) of the individuals in the study. The GRM is used to model the covariance structure of the genetic effects and the residuals in the linear mixed model. You can create the GRM file using the GCTA tool &quot;gcta --make-grm&quot;.</p></li>
<li><p>The trait file: This is a file containing the phenotype data of the individuals in the study. The trait file should be in the format of a text file (.phe), and should include the name of the trait and the phenotype values for each individual.</p></li>
<li><p>The MLM-LOCO command: To run MLM-LOCO, you need to specify the GCTA command line option &quot;--mlm&quot; followed by the option &quot;--loco&quot; to specify that MLM-LOCO should be used. You can also specify other options, such as the number of threads to use for parallel computing (&quot;-thread-num&quot;) and the significance threshold for genome-wide association (&quot;-p&quot;).</p></li>
<li><p>Here is an example command line to run MLM-LOCO in GCTA:</p></li>
</ul>

<div><pre><code class="language-none">gcta --mlm --bfile mydata --grm mygrm --pheno mytrait.phe --loco --out myresults</code></pre></div>

<ul>
<li>In this example, &quot;mydata&quot; is the name of the input binary PLINK file, &quot;mygrm&quot; is the name of the genetic relationship matrix file, &quot;mytrait.phe&quot; is the name of the phenotype file, and &quot;myresults&quot; is the name of the output file. The &quot;--loco&quot; option specifies that MLM-LOCO should be used. You can also add other options to customize the analysis as needed.</li>
</ul>
</blockquote>

<h3 id="toc_10">Q9. What tools do you recommend to perform power calculation and odds ratio calculations for a GWAS?</h3>

<blockquote>
<p>There are several tools and software packages available for performing power calculations and odds ratio calculations for a genome-wide association study (GWAS). Here are some recommendations:</p>

<ol>
<li>GPower: GPower is a free and user-friendly software package that can be used to perform power calculations for various types of statistical tests, including GWAS. GPower allows you to specify various parameters of your GWAS, such as sample size, effect size, alpha level, and statistical power, and provides estimates of the required sample size or power for your study. You can download GPower from the following website: https://gpower.software.informer.com/</li>
<li>QUANTO: QUANTO is another software package that can be used for power calculations in genetic association studies, including GWAS. QUANTO allows you to specify various parameters of your study, such as sample size, disease prevalence, effect size, and allele frequency, and provides estimates of the statistical power or required sample size. QUANTO is a commercial software package that can be purchased from the following website: http://biostats.usc.edu/software</li>
<li>R packages: There are several R packages that can be used to perform power calculations and odds ratio calculations for GWAS, including &quot;snpPower&quot;, &quot;powerGWASinteraction&quot;, and &quot;GWASTools&quot;. These packages provide functions for calculating power or sample size for various types of association tests, as well as functions for estimating odds ratios and other effect sizes. R is a free and open-source statistical software that can be downloaded from the following website: https://www.r-project.org/</li>
<li>Online calculators: There are also several online calculators and web-based tools that can be used for power calculations and odds ratio calculations for GWAS. Examples include &quot;GENESIS GWAS Power Calculator&quot; (https://genesis.bwh.harvard.edu/gwaspowercalc/) and &quot;POWER Calculator for Genetic Studies&quot; (http://zzz.bwh.harvard.edu/power/).</li>
</ol>

<p>These are just a few examples of the tools and software packages available for power calculations and odds ratio calculations in GWAS. The choice of tool will depend on your specific needs, the available resources, and your level of expertise.</p>
</blockquote>

<p>`</p>



<script type="text/javascript">
var _self="undefined"!=typeof window?window:"undefined"!=typeof WorkerGlobalScope&&self instanceof WorkerGlobalScope?self:{},Prism=function(){var e=/\blang(?:uage)?-(\w+)\b/i,t=0,n=_self.Prism={util:{encode:function(e){return e instanceof a?new a(e.type,n.util.encode(e.content),e.alias):"Array"===n.util.type(e)?e.map(n.util.encode):e.replace(/&/g,"&amp;").replace(/</g,"&lt;").replace(/\u00a0/g," ")},type:function(e){return Object.prototype.toString.call(e).match(/\[object (\w+)\]/)[1]},objId:function(e){return e.__id||Object.defineProperty(e,"__id",{value:++t}),e.__id},clone:function(e){var t=n.util.type(e);switch(t){case"Object":var a={};for(var r in e)e.hasOwnProperty(r)&&(a[r]=n.util.clone(e[r]));return a;case"Array":return e.map&&e.map(function(e){return n.util.clone(e)})}return e}},languages:{extend:function(e,t){var a=n.util.clone(n.languages[e]);for(var r in t)a[r]=t[r];return a},insertBefore:function(e,t,a,r){r=r||n.languages;var l=r[e];if(2==arguments.length){a=arguments[1];for(var i in a)a.hasOwnProperty(i)&&(l[i]=a[i]);return l}var o={};for(var s in l)if(l.hasOwnProperty(s)){if(s==t)for(var i in a)a.hasOwnProperty(i)&&(o[i]=a[i]);o[s]=l[s]}return n.languages.DFS(n.languages,function(t,n){n===r[e]&&t!=e&&(this[t]=o)}),r[e]=o},DFS:function(e,t,a,r){r=r||{};for(var l in e)e.hasOwnProperty(l)&&(t.call(e,l,e[l],a||l),"Object"!==n.util.type(e[l])||r[n.util.objId(e[l])]?"Array"!==n.util.type(e[l])||r[n.util.objId(e[l])]||(r[n.util.objId(e[l])]=!0,n.languages.DFS(e[l],t,l,r)):(r[n.util.objId(e[l])]=!0,n.languages.DFS(e[l],t,null,r)))}},plugins:{},highlightAll:function(e,t){var a={callback:t,selector:'code[class*="language-"], [class*="language-"] code, code[class*="lang-"], [class*="lang-"] code'};n.hooks.run("before-highlightall",a);for(var r,l=a.elements||document.querySelectorAll(a.selector),i=0;r=l[i++];)n.highlightElement(r,e===!0,a.callback)},highlightElement:function(t,a,r){for(var l,i,o=t;o&&!e.test(o.className);)o=o.parentNode;o&&(l=(o.className.match(e)||[,""])[1],i=n.languages[l]),t.className=t.className.replace(e,"").replace(/\s+/g," ")+" language-"+l,o=t.parentNode,/pre/i.test(o.nodeName)&&(o.className=o.className.replace(e,"").replace(/\s+/g," ")+" language-"+l);var s=t.textContent,u={element:t,language:l,grammar:i,code:s};if(!s||!i)return n.hooks.run("complete",u),void 0;if(n.hooks.run("before-highlight",u),a&&_self.Worker){var c=new Worker(n.filename);c.onmessage=function(e){u.highlightedCode=e.data,n.hooks.run("before-insert",u),u.element.innerHTML=u.highlightedCode,r&&r.call(u.element),n.hooks.run("after-highlight",u),n.hooks.run("complete",u)},c.postMessage(JSON.stringify({language:u.language,code:u.code,immediateClose:!0}))}else u.highlightedCode=n.highlight(u.code,u.grammar,u.language),n.hooks.run("before-insert",u),u.element.innerHTML=u.highlightedCode,r&&r.call(t),n.hooks.run("after-highlight",u),n.hooks.run("complete",u)},highlight:function(e,t,r){var l=n.tokenize(e,t);return a.stringify(n.util.encode(l),r)},tokenize:function(e,t){var a=n.Token,r=[e],l=t.rest;if(l){for(var i in l)t[i]=l[i];delete t.rest}e:for(var i in t)if(t.hasOwnProperty(i)&&t[i]){var o=t[i];o="Array"===n.util.type(o)?o:[o];for(var s=0;s<o.length;++s){var u=o[s],c=u.inside,g=!!u.lookbehind,h=!!u.greedy,f=0,d=u.alias;u=u.pattern||u;for(var p=0;p<r.length;p++){var m=r[p];if(r.length>e.length)break e;if(!(m instanceof a)){u.lastIndex=0;var y=u.exec(m),v=1;if(!y&&h&&p!=r.length-1){var b=r[p+1].matchedStr||r[p+1],k=m+b;if(p<r.length-2&&(k+=r[p+2].matchedStr||r[p+2]),u.lastIndex=0,y=u.exec(k),!y)continue;var w=y.index+(g?y[1].length:0);if(w>=m.length)continue;var _=y.index+y[0].length,P=m.length+b.length;if(v=3,P>=_){if(r[p+1].greedy)continue;v=2,k=k.slice(0,P)}m=k}if(y){g&&(f=y[1].length);var w=y.index+f,y=y[0].slice(f),_=w+y.length,S=m.slice(0,w),O=m.slice(_),j=[p,v];S&&j.push(S);var A=new a(i,c?n.tokenize(y,c):y,d,y,h);j.push(A),O&&j.push(O),Array.prototype.splice.apply(r,j)}}}}}return r},hooks:{all:{},add:function(e,t){var a=n.hooks.all;a[e]=a[e]||[],a[e].push(t)},run:function(e,t){var a=n.hooks.all[e];if(a&&a.length)for(var r,l=0;r=a[l++];)r(t)}}},a=n.Token=function(e,t,n,a,r){this.type=e,this.content=t,this.alias=n,this.matchedStr=a||null,this.greedy=!!r};if(a.stringify=function(e,t,r){if("string"==typeof e)return e;if("Array"===n.util.type(e))return e.map(function(n){return a.stringify(n,t,e)}).join("");var l={type:e.type,content:a.stringify(e.content,t,r),tag:"span",classes:["token",e.type],attributes:{},language:t,parent:r};if("comment"==l.type&&(l.attributes.spellcheck="true"),e.alias){var i="Array"===n.util.type(e.alias)?e.alias:[e.alias];Array.prototype.push.apply(l.classes,i)}n.hooks.run("wrap",l);var o="";for(var s in l.attributes)o+=(o?" ":"")+s+'="'+(l.attributes[s]||"")+'"';return"<"+l.tag+' class="'+l.classes.join(" ")+'" '+o+">"+l.content+"</"+l.tag+">"},!_self.document)return _self.addEventListener?(_self.addEventListener("message",function(e){var t=JSON.parse(e.data),a=t.language,r=t.code,l=t.immediateClose;_self.postMessage(n.highlight(r,n.languages[a],a)),l&&_self.close()},!1),_self.Prism):_self.Prism;var r=document.currentScript||[].slice.call(document.getElementsByTagName("script")).pop();return r&&(n.filename=r.src,document.addEventListener&&!r.hasAttribute("data-manual")&&document.addEventListener("DOMContentLoaded",n.highlightAll)),_self.Prism}();"undefined"!=typeof module&&module.exports&&(module.exports=Prism),"undefined"!=typeof global&&(global.Prism=Prism);
</script>


</body>

</html>
